### PR TITLE
docs(store): add note about using createFeatureSelector with no generic supplied to Store

### LINKED ISSUE #2780: select overload not match when enable strict mode in tsconfig
I create a angular project with **strict** flag and try add in a feature store. But when I try to select the state with selector have created it appear overload not match. When I turn off  **strictFunctionTypes** everything work well.

## Minimal reproduction of the bug/regression with instructions:
[Repo URL](https://github.com/hanjeahwan/ngrx-strict-mode)

## Expected behavior:
No error

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):
NgRx : 10.0.1
Angular: 10.1.4

### PR DESCRIPTION
**Pull request by david-shortman**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #2780

## What is the new behavior?

Added doc explaining usage of `createFeatureSelector` with `Store`

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/store/selectors.md [hunk 1]
```diff
@@ -266,6 +266,27 @@ export class AppComponent {
...
+<code-example header="index.ts">
+import { createSelector, createFeatureSelector } from '@ngrx/store';
+
+export const featureKey = 'feature';
+
+export interface FeatureState {
+  counter: number;
+}
+
+export const selectFeature = createFeatureSelector<FeatureState>(featureKey);
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/store/selectors.md:282]: The arrows for the type have to be escaped.

```suggestion
export const selectFeature = createFeatureSelector&lt;FeatureState&gt;(featureKey);
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: projects/ngrx.io/content/guide/store/selectors.md [hunk 2]
```diff
@@ -266,6 +266,27 @@ export class AppComponent {
 }
 </code-example>
 
+When omitting a generic for the Store, only selectors which operate on `object`s are allowed in `store.select`.
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/store/selectors.md:269]: That's true, but only for strict mode.

File: projects/ngrx.io/content/guide/store/selectors.md [hunk 3]
```diff
@@ -266,6 +266,27 @@ export class AppComponent {
...
+
+export interface FeatureState {
+  counter: number;
+}
+
+export const selectFeature = createFeatureSelector<FeatureState>(featureKey);
+
+export const selectFeatureCount = createSelector(
+  selectFeature,
+  (state: FeatureState) => state.counter
```
Comment (markostanimirovic) [inline projects/ngrx.io/content/guide/store/selectors.md:286]: the type of `state` will be inferred from the feature selector

```suggestion
  (state) => state.counter
```

Comment (david-shortman) [reply projects/ngrx.io/content/guide/store/selectors.md:286]: yeah, the many of the examples in this document needlessly explicitly declare the type of projector arguments. i can remove it- i was unsure.

File: projects/ngrx.io/content/guide/store/selectors.md [hunk 4]
```diff
@@ -248,15 +248,15 @@ selectTotal.release();
 
 ## Using Store Without Type Generic
 
-When injecting `Store` into components and other injectables, it is possible to omit the generic type. If injected without the generic, the default generic is applied as follows `Store<T = object>`.
+The most common way to select information from the store is to use a selector function defined with `createSelector`. TypeScript is able to automatically infer types from `createSelector`, which reduces the need to provide the shape of the state to `Store` via a generic argument.
 
-The most common way to select information from the store is to use a selector function defined with `createSelector`. When doing so, TypeScript is able to automatically infer types from the selector function, therefore reducing the need to define the type in the store generic.
+So, when injecting `Store` into components and other injectables, the generic type can be omitted. If injected without the generic, the default generic applied is `Store<T = object>`.
```
Comment (david-shortman) [inline projects/ngrx.io/content/guide/store/selectors.md:253]: - removed use of passive voice
- placed motivation to use `Store` without type generic at beginning of section

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**